modules/core/src/testFixtures/java/org/apache/ignite/internal/testframework/log4j2/TestLogChecker.java

```diff
@@ -0,0 +1,324 @@
+        TestLogChecker checker = new TestLogChecker(loggerName);
+
+        if (started) {
+            checker.start();
+        }
+
+        return checker;
+    }
+
+    /**
+     * Creates a new instance of {@link TestLogChecker} for the given {@code clazz}.
+     *
+     * @return New instance of {@link TestLogChecker}.
+     **/
+    public static TestLogChecker create(Class<?> clazz) {
+        return create(clazz, false);
+    }
+
+    /**
+     * Creates a new instance of {@link TestLogChecker} for the given {@code clazz}.
+     *
+     * @param started Whether the checker should be started.
+     * @return New instance of {@link TestLogChecker}.
+     **/
+    public static TestLogChecker create(Class<?> clazz, boolean started) {
+        return create(clazz.getName(), started);
+    }
+
+    /**
+     * Creates a new instance of {@link TestLogChecker}.
+     *
+     * @param loggerName Logger name.
+     */
+    public TestLogChecker(String loggerName) {
+        Objects.requireNonNull(loggerName);
+
+        this.loggerName = loggerName;
+        this.appender = new TestLogAppender(loggerName);
+    }
+
+    /**
+     * Creates a new instance of {@link TestLogChecker} with the given {@code predicate}.
+     *
+     * @param loggerName Logger name.
+     * @param predicate Predicate to check log messages.
+     */
+    public TestLogChecker(String loggerName, Predicate<String> predicate) {
+        this(loggerName, predicate, () -> {});
+    }
+
+    /**
+     * Creates a new instance of {@link TestLogChecker} with the given {@code predicate} and {@code action}.
+     *
+     * @param loggerName Logger name.
+     * @param predicate Predicate to check log messages.
+     * @param action Action to be executed when the {@code predicate} is matched.
+     */
+    public TestLogChecker(String loggerName, Predicate<String> predicate, Runnable action) {
+        this(loggerName, new Handler(predicate, action));
+    }
+
+    /**
+     * Creates a new instance of {@link TestLogChecker} with the given {@code predicate} and {@code action}.
+     *
+     * @param loggerName Logger name.
+     * @param handlers List of handlers to be added.
+     */
+    public TestLogChecker(String loggerName, Handler... handlers) {
+        this(loggerName);
+
+        lock.writeLock().lock();
+        try {
+            Collections.addAll(this.handlers, handlers);
+        } finally {
+            lock.writeLock().unlock();
+        }
+    }
+
+    /**
+     * Adds a new handler with the given {@code predicate} and {@code action}.
+     *
+     * @param predicate Predicate to check log messages.
+     * @param action Action to be executed when the {@code predicate} is matched.
+     * @return New instance of {@link Handler}.
+     */
+    public Handler addHandler(Predicate<String> predicate, Runnable action) {
+        Handler handler = new Handler(predicate, action);
+
+        lock.writeLock().lock();
```

Comment:
It seems we can just invoke `addHandler(handler)` here

Reply:
Thanks. Fixed.